### DISCUSSION TITLE
PCB Drilling / G-Code orientation (Transformation Matrix)

### DISCUSSION THREAD

**Discussion by jkirk** [vlachoudis/bCNC#1842] [General]
Hi!

First and foremost: Thank you for this software! ❤️ 

I am new to CNC in general and bCNC in particular, so please bear with me.

We are trying to drill some PCBs and we had a hard time figuring out how to align the PCB with our CNC.
We found this [Video (Using Transform Matrix for PCB Drilling)](https://youtu.be/tzI445XsPDs) and wanted to do the same with bCNC.
Basically we want to do the following:

- Configure / the camera / webcam
- Determine the camera offset with a reference hole
- Orient / Rotate the G-Code (via a transformation matrix)

If found and tried to follow [Probe Camera Alignment](https://github.com/vlachoudis/bCNC/wiki/Probe-Camera-Alignment), which seems to be a bit out of date  (i.e. there is no "Tools -> Config -> Camera") + [Tutorials: Orientation of gcode](https://github.com/vlachoudis/bCNC/wiki/Tutorials%3A-Orientation-of-gcode), but I still have a few (newbie) questions:

- Probe Camera Alignment

  - How to configure the camera? I think I figured this one out: CAM -> Camera -> set Align Camera (+ Web Camera?).
But: what is the difference between "Align Camera" + "Web Camera" (where are they used)?
  How to apply the configuration? (I had to restart bCNC to apply the change.)
The rotation is set in "Probe -> Camera -> Rotation", right?
  - **Diameter** mentioned in the Wiki is actually **Crosshair**, right?
  - Can I just update the "Configuration" section in the Wiki?
- Orientation of G-Code

  - What does "_Switch to the camera system, by clicking the "Switch" button on the ribbon._" mean? Is this just "Probe -> Camera" or the "Toggle display of camera" above the workspace or is it the "Probe -> Switch To" button?
  - What does "_The button creates a virtual offset with G92 on the present workspace as the center is moved to the camera location._" mean? What is G92? Where can I see it? I found G92 in "Control -> State" below the workspaces "G54" .. "G59". Is this it? Why are the workspaces called "State" / GXX? Can you point me to some documentation about this?

Thanks for any help!

**Comment by BlakusSenior** [comment]
Not sure what program you use but I find 'Flatcam' a free program to write the cnc code most useful. You simply load a gerber, drill code and it will create drilling codes for you. All of which can be seen on a visual display of the project.
